src/north/ApiHandler.class.js

```diff
@@ -65,6 +89,18 @@ class ApiHandler {
   disconnect() {
     const { name, id } = this.application
     this.logger.info(`North API ${name} (${id}) disconnected`)
+    this.engine.eventEmitters[`/north/${id}/sse`]?.events?.off('data', this.listener)
+    this.statusData['Connected at'] = 'Not connected'
+    this.updateStatusDataStream()
+  }
+
+  listener = (data) => {
+    if (data) this.sseData = data
+    this.engine.eventEmitters[`/north/${this.application.id}/sse`]?.stream?.write(`data: ${JSON.stringify(this.sseData)}\n\n`)
```

Comment:
Why is it different than protocolHandler ? You don't need sse and you should check if data exists (see listener method in ProtocolHandler.class.js)